sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/async_challenge_auth_policy.py

```diff
@@ -86,7 +86,13 @@ async def on_challenge(self, request: "PipelineRequest", response: "PipelineResp
 
         body = request.context.pop("key_vault_request_data", None)
         request.http_request.set_text_body(body)  # no-op when text is None
-        await self.authorize_request(request, scope, tenant_id=challenge.tenant_id)
+
+        # The tenant parsed from AD FS challenges is "adfs"; we don't actually need a tenant for AD FS authentication
+        # For AD FS we skip cross-tenant authentication per https://github.com/Azure/azure-sdk-for-python/issues/28648
+        if challenge.tenant_id.lower().endswith("adfs"):
```

Comment:
Is it possible to have "adfs/"?

Reply:
Good point; but that won't happen because of how the `tenant_id` is parsed: https://github.com/Azure/azure-sdk-for-python/blob/5e12f3fee180b4b62069a2ce2f06f5b63a50aa85/sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/http_challenge.py#L50